lms/djangoapps/grades/signals/handlers.py

```diff
@@ -124,19 +109,53 @@ def recalculate_subsection_grade_handler(sender, **kwargs):  # pylint: disable=u
     """
     if not settings.FEATURES.get('ENABLE_SUBSECTION_GRADES_SAVED', False):
         return
-    try:
-        course_id = kwargs.get('course_id', None)
-        usage_id = kwargs.get('usage_id', None)
-        student = kwargs.get('user', None)
 
-        course_key = CourseLocator.from_string(course_id)
-        usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
-
-        from lms.djangoapps.grades.new.subsection_grade import SubsectionGradeFactory
-        SubsectionGradeFactory(student).update(usage_key, course_key)
-    except Exception as ex:  # pylint: disable=broad-except
+    try:
+        course_id = kwargs['course_id']
+        usage_id = kwargs['usage_id']
+        student = kwargs['user']
+    except KeyError:
         log.exception(
-            u"Failed to process SCORE_CHANGED signal. "
-            "user: %s, course_id: %s, "
-            "usage_id: %s. Exception: %s", unicode(student), course_id, usage_id, ex.message
+            u"Failed to process SCORE_CHANGED signal, some arguments were missing."
+            "user: %s, course_id: %s, usage_id: %s.",
+            kwargs.get('user', None),
+            kwargs.get('course_id', None),
+            kwargs.get('usage_id', None),
         )
+        return
+
+    course_key = CourseLocator.from_string(course_id)
+    usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
+    manager = get_block_structure_manager(course_key)
+    block_structure = manager.get_collected()
```

Comment:
Instead of the above 2 lines, call `openedx.core.djangoapps.content.block_structure.api.get_course_in_cache` instead.